src/Microsoft.AspNet.Authentication.OAuthBearer/OAuthBearerAuthenticationOptions.cs

```diff
@@ -103,53 +93,25 @@ public OAuthBearerAuthenticationOptions() : base()
         /// Gets or sets if a metadata refresh should be attempted after a SecurityTokenSignatureKeyNotFoundException. This allows for automatic
         /// recovery in the event of a signature key rollover. This is enabled by default.
         /// </summary>
-        public bool RefreshOnIssuerKeyNotFound { get; set; }
+        public bool RefreshOnIssuerKeyNotFound { get; set; } = true;
 
         /// <summary>
         /// Used to know what the current clock time is when calculating or validating token expiration. When not assigned default is based on
         /// DateTimeOffset.UtcNow. This is typically needed only for unit testing.
         /// </summary>
-        public ISystemClock SystemClock { get; set; }
+        public ISystemClock SystemClock { get; set; } = new SystemClock();
 
         /// <summary>
         /// Gets or sets the <see cref="SecurityTokenValidators"/> for validating tokens.
         /// </summary>
         /// <exception cref="ArgumentNullException">if 'value' is null.</exception>
-        public ICollection<ISecurityTokenValidator> SecurityTokenValidators
-        {
-            get
-            {
-                return _securityTokenValidators;
-            }
-
-            [param: NotNull]
-            set
-            {
-                _securityTokenValidators = value;
-            }
-        }
+        public IList<ISecurityTokenValidator> SecurityTokenValidators { get; } = new List<ISecurityTokenValidator>();
 
         /// <summary>
         /// Gets or sets the TokenValidationParameters
         /// </summary>
         /// <remarks>Contains the types and definitions required for validating a token.</remarks>
         /// <exception cref="ArgumentNullException">if 'value' is null.</exception>
-        public TokenValidationParameters TokenValidationParameters
-        {
-            get
-            {
-                return _tokenValidationParameters;
-            }
-
-            set
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException("TokenValidationParameters");
-                }
-
-                _tokenValidationParameters = value;
-            }
-        }
+        public TokenValidationParameters TokenValidationParameters { get; } = new TokenValidationParameters();
```

Comment:
@PinpointTownes The issue on Github is not related to the ability to set the TVP. It is about the ordering problem. That issue can occur even without the setter if user sets TVP.AuthenticationType after setting the SetDefaultSignInAsAuthenticationType.Also, that issue does not exist in vNext (as far as I can tell) as we do not set the SignInScheme the same way we were setting the SetDefaultSignInAsAuthenticationType